### PR TITLE
Patch Fix for base GATK - upgrade to TF2.6

### PR DESCRIPTION
**Pull request by ravi9**
@rtitle @xuscrbw  I created a patch to fix the TF2.6 upgrade issue. Also, related: #258.

- Updated Dockerfile
   - Installed GATK Python module
   - Added a patch fix to fix TF2.6 compatibility issues. No need to rebuild GATK.
   
 - Updated gatk_smoke_test.ipynb. I tested locally and it works fine.

### CODE REVIEW

File: terra-jupyter-gatk/gatk-tf.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
+--- /opt/conda/lib/python3.7/site-packages/vqsr_cnn/vqsr_cnn/models.py  2021-09-23 22:44:09.000000000 +0000
++++ models.py   2021-09-23 22:48:06.197472317 +0000
+@@ -5,7 +5,7 @@
+ from keras import layers
+ from keras import metrics
+ import keras.backend as K
+-from keras.optimizers import Adam
++from keras.optimizers import adam_v2 as Adam
```
Comment (Qi77Qi) [inline terra-jupyter-gatk/gatk-tf.patch:8]: does it make sense to PR in gatk repo directly?

**Review by Qi77Qi — COMMENTED** [review]
thanks @ravi9 ! Looks like the `terra-jupyter-gatk` test failed, but that's probably because the ci doesn't work well for forks...

cc @xuscrbw since you have a similar PR open

File: terra-jupyter-gatk/Dockerfile [hunk 2]
```diff
@@ -34,6 +34,12 @@ RUN curl -L -o $GATK_ZIP_PATH https://github.com/broadinstitute/gatk/releases/do
  && unzip -o $GATK_ZIP_PATH -d /etc/ \
  && ln -s /etc/gatk-$GATK_VERSION/gatk /bin/gatk
 
+ENV PIP_USER=false
+RUN pip install /etc/gatk-4.2.0.0/gatkPythonPackageArchive.zip
```
[the remark this answers is not in the document]
Comment (rtitle) [reply terra-jupyter-gatk/Dockerfile:38]: Maybe could say `/etc/gatk-$GATK_VERSION/..` just so everything is controlled by the env var

File: terra-jupyter-gatk/Dockerfile [hunk 3]
```diff
@@ -34,6 +34,12 @@ RUN curl -L -o $GATK_ZIP_PATH https://github.com/broadinstitute/gatk/releases/do
...
+ENV PIP_USER=true
+
+COPY gatk-tf.patch /etc/gatk-4.2.0.0/gatk-tf.patch
```
Comment (rtitle) [inline terra-jupyter-gatk/Dockerfile:41]: Same as above

**Comment by rtitle** [comment]
This looks good to me too.. @xuscrbw maybe you could merge this into https://github.com/DataBiosphere/terra-docker/pull/258/ then merge that PR to main. Downloading the gatk zip from github seems fine to me if it works.

File: terra-jupyter-gatk/gatk-tf.patch (hunk 1 above)
Comment (ravi9) [reply terra-jupyter-gatk/gatk-tf.patch:8]: I'm not sure. This is a hotfix for the TF upgrade issue. We can suggest this to the GATK maintainers. I assume they are fixing this in the next GATK release!

**Comment by ravi9** [comment]
Yes, downloading the zip works fine.

**Comment by rtitle** [comment]
Closing in favor of https://github.com/DataBiosphere/terra-docker/pull/258

**Pull request closed without merging**